R/helper_clean_model_data.R

```diff
@@ -85,25 +101,25 @@ clean_model_data <- function(formula,
     ret[["original_treatment"]] <- mf[, colnames(mf) == all.vars(terms(mf)[[3]])[1]]
   }
 
-  if (!missing(weights)) {
+  if (!is.null(mfargs$weights)) {
```

Comment:
is.numeric

Reply:
Actually, no on this one because the arg isn't the weights itself, its just the name of the weights object that is evaluated above.

Reply:
I rescind my thumbs up!

Reply:
If weights is not in the model.frame, it returns NULL, and    
```
  ret[["weights"]] <- model.extract(mf, "weights")
```
still works unconditionally.

Reply:
```
> ret <- list()
> mf <- model.frame(extra~group, data=sleep)
> ret[["weights"]] <- model.extract(mf, "weights")
> ret
list()
> any(ret[["weights"]] < 0)
[1] FALSE
```